src/Core/src/Platform/iOS/MauiLabel.cs

```diff
@@ -83,7 +83,15 @@ public override void InvalidateIntrinsicContentSize()
 			}
 		}
 
-		public override SizeF SizeThatFits(SizeF size) => AddInsets(base.SizeThatFits(size));
+		public override SizeF SizeThatFits(SizeF size)
+		{
+			var requestedSize = base.SizeThatFits(size);
+
+			// Let's be sure the label is not larger than the container
```

Comment:
I believe this will break Labels with `TextWrapping` set to `NoWrap`; it will cut them off in the middle, instead of allowing them to be laid out past the edge of the container. 

So this probably needs to check whether the UILabel's `LineBreakMode` is one of the truncation options, and default to the old implementation if not. 

And as far as I can tell, the base SizeThatFits() implementation only does this weirdness when the `Lines` property is set to 1 (which we do in Controls when setting one of the truncation options for compatibility reasons). So it likely needs to check whether truncation is enabled and the `Lines` property is set to 1. Otherwise, fall back to the old implementation.

Reply:
Good call! I added the conditionals and added those into the tests to make sure that the label can exceed the parent container when it has a HorizontalOption that is not the default fill and has the NoWrap LineBreakMode.

It may be worth noting that a label with `LineBreakMode=NoWrap` and the default `HorizontalOption=fill`, will still stay at the size of the parent container. This is the expected behavior here right?

Reply:
@hartez, actually, based on the test failures, it seems that on android, NoWrap does not exceed the parent container like iOS is currently doing. Which is the preferred behavior?

Reply:
@hartez ^ 

With the following yaml:
```yml
    <VerticalStackLayout WidthRequest="200" Spacing="20" Background="LightBlue">
        <Label HorizontalOptions="Start" LineBreakMode="NoWrap" Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />
        <Label LineBreakMode="NoWrap" Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />
    </VerticalStackLayout>
```

![image](https://user-images.githubusercontent.com/50846373/233407066-a880f0ca-f2c2-448b-916f-8a6c73747b71.png)

![image](https://user-images.githubusercontent.com/50846373/233410711-9a8b3219-64f3-40ba-a41f-2fb4e41ade6e.png)

Reply:
Ios feels wrong, right? The reason I have a layout is to contain the contents. What case will we want the label to exceed the bounds of the container on just one platform?

Reply:
> The reason I have a layout is to contain the contents.

Layouts don't _have to_ clip their content; there's an option for that (IsClippedToBounds). So content exceeding the bounds of a layout is not a crazy idea. 

And I'd argue that the behavior of Windows/Android seems wrong to me here, but it's also the publicly released behavior so we're stuck with it even if I don't like it. :)

So TJ, ignore my objections, the original code was fine.

Reply:
@hartez okay, updated!